R/get_custom_effort.R

```diff
@@ -231,33 +237,12 @@ get_custom_effort <- function(package = NULL,
         effort = sum(.data$sum_effort, na.rm = TRUE)
       )
   } else {
-    if (group_by == "day") {
-      period <- 1 # ndays within a group by unit
-    }
-
-    if (group_by == "week") {
-      period <- 7 # ndays within a group by unit
-    }
-    if (group_by == "month") {
-      period <- 30 # ndays within a group by unit
-    }
-    if (group_by == "year") {
-      period <- 365 # ndays within a group by unit
-    }
-    # add period column and group by it
     sum_effort <-
       sum_effort %>%
-      dplyr::mutate(period = as.numeric(.data$date - .data$date[1]) %/% period) %>%
-      dplyr::group_by(.data$period)
-
-    # sum total effort over each interval
-    sum_effort <-
-      sum_effort %>%
-      dplyr::summarise(
-        begin = min(.data$date, na.rm = TRUE),
-        effort = sum(.data$sum_effort, na.rm = TRUE)
-      ) %>%
-      dplyr::ungroup()
+      dplyr::mutate(
+        begin = lubridate::floor_date(.data$date, unit = group_by)) %>%
+      dplyr::group_by(begin) %>%
```

Comment:
`no visible binding for global variable ‘begin’`

```suggestion
      dplyr::group_by(.data$begin) %>%
```

Reply:
Yes, I forgot it! Thanks.